lmcache/experimental/storage_backend/local_disk_backend.py

```diff
@@ -0,0 +1,159 @@
+        self.dict: OrderedDict[CacheEngineKey,
+                               DiskCacheMetadata] = OrderedDict()
+        self.dst_device = dst_device
+
+        # TODO: Tune the number of workers for better performance
+        self.proc_pool_executor = ProcessPoolExecutor(max_workers=4)
+
+        self.disk_lock = threading.Lock()
+        assert config.local_disk is not None
+        self.path: str = config.local_disk
+        if not os.path.exists(self.path):
+            os.makedirs(self.path)
+            logger.info(f"Created local disk cache directory: {self.path}")
+
+        # TODO(Jiayi): Size and evictor should be configured
+
+        self.is_shutdown = False
+
+    def __str__(self):
+        return self.__class__.__name__
+
+    def contains(self, key: CacheEngineKey) -> bool:
+        with self.disk_lock:
+            return key in self.dict
+
+    def _key_to_path(
+        self,
+        key: CacheEngineKey,
+    ) -> str:
+        return self.path + key.to_string().replace("/", "-") + ".pt"
+
+    def insert_key(self, key: CacheEngineKey, size: int):
+        path = self._key_to_path(key)
+        with self.disk_lock:
+            self.dict[key] = DiskCacheMetadata(path, size)
+
+    def submit_put_task(
+        self,
+        key: CacheEngineKey,
+        memory_obj: MemoryObj,
+    ) -> Future:
+        kv_chunk = memory_obj.tensor
+        fmt_str = str(memory_obj.metadata.fmt.value)
+        path = self._key_to_path(key)
+        future = self.proc_pool_executor.submit(save_disk, path, kv_chunk,
+                                                fmt_str)
+        return future
+
+    def put_blocking(
+        self,
+        key: CacheEngineKey,
+        memory_obj: MemoryObj,
+    ) -> None:
+        """
+        Blocking put function.
+        This is for debugging and testing purposes.
+        """
+        path = self._key_to_path(key)
+        kv_chunk = memory_obj.tensor
+        fmt_str = str(memory_obj.metadata.fmt.value)
+        save_disk(path, kv_chunk, fmt_str)
+        self.insert_key(key, memory_obj.get_size())
+
+    def submit_prefetch_task(
+        self,
+        key: CacheEngineKey,
+    ) -> Optional[Future]:
+
+        self.disk_lock.acquire()
+        if key not in self.dict:
+            self.disk_lock.release()
+            return None
+        path = self.dict[key].path
+        self.disk_lock.release()
+        logger.info(f"Prefetching {key} from disk.")
+        future = self.proc_pool_executor.submit(load_disk, path, "cpu")
+        return future
+
+    def get_blocking(
+        self,
+        key: CacheEngineKey,
+    ) -> Optional[BufferMemoryObj]:
+        """
+        Blocking get function.
+        """
+        self.disk_lock.acquire()
+        if key not in self.dict:
+            self.disk_lock.release()
+            return None
+        path = self.dict[key].path
+        buffer_memory_obj = load_disk(path, self.dst_device)
+        self.disk_lock.release()
```

Comment:
Should we do `self.disk_lock.release()` before calling the `load_disk`? Otherwise, this function may hold the lock for a very long time when reading the tensor from the disk.

Reply:
Not sure if it's a good idea. I intentionally lock this since I want to prioritize `load_disk` over `save_disk`. Multiple concurrent/parallel `load_disk` could increase throughput but it's a blocking function for now anyway.